### PR TITLE
#64 new design header layout

### PR DESCRIPTION
**Pull request by baxgas**
Made the home page layout according to Csenge's Designs, not all parts are finished, but if you approve it, this could be demoed to the UX/UI team

### CODE REVIEW

File: pages/index.vue [hunk 1]
```diff
@@ -1,23 +1,10 @@
...
+  <div>
+    <section class="grid-cols-2 mb-16 md:grid">
       <div>
-        <nuxt-link to="news/">
-          <h3>News</h3>
-        </nuxt-link>
-        <div v-for="news in sortNews" :key="news.id">
-          <NewsBlock :news="news" />
-        </div>
+        <NewsHome :news="sortNews[newsStart]" @changenews="changenews" />
```
Comment (gammaw) [inline pages/index.vue:5]: Wow, didn't know that you can refer to the /news/Home.vue component like this. 
(I found it also in the docs: https://nuxtjs.org/docs/2.x/directory-structure/components/)

**Review by gammaw — COMMENTED** [review]
Seems to be a nice structure! I guess I got the big picture. 
Added a question and comment. 

Thanks a lot for the this big chunk of work!

File: components/HeaderBlock.vue [hunk 2]
```diff
@@ -1,13 +1,20 @@
 <template>
-  <header class="container flex-row-reverse py-4 md:flex md:flex-row">
-    <div>
-      <client-only>
-        <RadioPlayer :show-album-art="true" :now-playing-uri="streamServer" />
-      </client-only>
+  <header class="flex-row-reverse items-center justify-between px-4 py-4 bg-white md:flex md:flex-row">
```
Comment (pvj) [inline components/HeaderBlock.vue:2]: Wow! I learned a lot looking these classes up.

File: components/HeaderBlock.vue [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+    <div class="flex items-center">
+      <div class="block mr-4 cursor-pointer" @click="$router.push('/')">
+        <img src="/img/lahmacun-logo.png" alt="" class="w-24">
+      </div>
+      <div class="mr-8">
+        <client-only>
+          <RadioPlayer :show-album-art="true" :now-playing-uri="streamServer" />
```
Comment (pvj) [inline components/HeaderBlock.vue:9]: The `uri` part is really cool. Does this set-up mean we could - technically - easily use the same player for `streamServer` and then on click for `arcsi` and then back on streamServer. 

Maybe this is not the right place to discuss it, commenting on 1 line of code in the middle of PR but I'm thinking lot about the following: 

Let's say we have only 1 unified player. By default it plays the broadcast stream. User arrives and clicks on an arcsived episode to listen. This is fairly clear where the button is to listen to an old episode. The player switches from broadcast to playback. Here's my question: How/where can you switch back to listen to the radio? I've been thinking about this for some time and this is not clear to me.

File: components/HeaderBlock.vue [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
+        </client-only>
+      </div>
     </div>
-    <div class="flex items-center cursor-pointer" @click="$router.push('/')">
-      <img src="/img/lahmacun-logo.png" alt="" class="h-16 mr-4">
-      <h1>Lahmacun Radio</h1>
+    <HeadNavigation />
+    <div id="time">
+      <div class="currenttime">
+        {{ timestamp }}
```
Comment (pvj) [inline components/HeaderBlock.vue:16]: I know you are going by the design but would it be possible to add two timezones? One is your computer's time one is the lahmacun time (UTC+2).

File: components/HeaderBlock.vue [hunk 5]
```diff
@@ -22,15 +29,25 @@ export default {
...
-      streamServer
+      streamServer,
+      timestamp: ''
     }
   },
   computed: {
 
   },
+  created () {
+    setInterval(this.getNow, 1000)
```
Comment (pvj) [inline components/HeaderBlock.vue:40]: nice!

File: components/HeaderBlock.vue [hunk 6]
```diff
@@ -22,15 +29,25 @@ export default {
...
+  },
+
   mounted () {
   },
   methods: {
+    getNow () {
```
Comment (pvj) [inline components/HeaderBlock.vue:46]: Question for the future: 
the `getNow()` function might be global function called on different multiple components, should we move it to some other place? 

Perhaps we will add /schedule page where we display crosshair to show the user where we are in the program. That crosshair could use the exact same method (`getNow()`) and exact same logic (`setInterval(this.getNow, 1000)`)

File: components/RadioPlayer.vue [hunk 7]
```diff
@@ -7,6 +7,14 @@
 
       <div class="now-playing-details">
         <div class="radio-controls">
+          <div v-if="showAlbumArt && np.now_playing.song.art" class="now-playing-art">
```
Comment (pvj) [inline components/RadioPlayer.vue:10]: is `showAlbumArt` settable somewhere? Could we maybe create a config for the player with list of such options? 

- show album art
- display elapsed time
- display remaining time
- display chatbox
- display `ON AIR` banner

etc. 

That could be really useful for all the different types of things we will be using the player for later (eg. live shows, repeats, shared broadcasts, one-off events etc.)

File: components/RadioPlayer.vue [hunk 8]
```diff
@@ -7,6 +7,14 @@
...
+            <a class="cursor-pointer programimage" rel="playerimg" @click.stop="streamModal = !streamModal">
+              <div v-if="show_check == true" class="onair">On air</div>
+              <img class="progimg" :src="show_art_url" :alt="'album_art_alt'">
+            </a>
+            <Modal :media="show_art_url" :title="show_title" :description="show_subtitle" :visibility="streamModal" @close="closeModal" />
```
Comment (pvj) [inline components/RadioPlayer.vue:15]: What happens on the `@close` action?

File: components/RadioPlayer.vue [hunk 9]
```diff
@@ -7,6 +7,14 @@
...
+          </div>
+
           <div class="play-volume-controls">
```
Comment (pvj) [inline components/RadioPlayer.vue:18]: This is again maybe the wrong place -- let me know if we should move this discussion just say please create a separate issue for this, dude and I will do so -- 

Do you think we should add skip-backward skip-forward buttons? Especially for arcsi items that would be useful to be able to use them instead of seeking by hand

File: components/RadioPlayer.vue [hunk 10]
```diff
@@ -22,41 +30,33 @@
             <a href="#" class="text-secondary volumeshower" @mouseenter="showVolumeSlider = !showVolumeSlider">
```
Comment (pvj) [inline components/RadioPlayer.vue:30]: How come these are not part of the player modal? Maybe I'm misunderstanding what the Modal here means or what the role of the Modal is? Can  you link me to some place where I may dive deeper?

File: components/RadioPlayer.vue [hunk 11]
```diff
@@ -22,41 +30,33 @@
...
-                step="1"
-              >
-            </div>
-            <div class="radio-control-max-volume-button">
-              <a href="#" class="text-secondary" :title="'full_volume_btn'" @click.prevent="volume = 100">
-                <i class="material-icons" aria-hidden="true">volume_up</i>
-              </a>
+            <div v-show="showVolumeSlider" id="radio-player-controls" class="radio-controls-standalone volumecontrolos" @mouseleave="showVolumeSlider = !showVolumeSlider">
+              <div class="radio-control-mute-button">
+                <a href="#" class="text-secondary" :title="'mute_btn'" @click.prevent="volume = 0">
```
Comment (pvj) [inline components/RadioPlayer.vue:36]: Does here `@click.prevent` event mean that it prevents clicking _if_ the volume is already 0? 

Great that we are adding mute functionality! This was 1 thing that people asked about a lot!!

File: components/HeaderBlock.vue [hunk 12]
```diff
@@ -1,13 +1,20 @@
...
+        <client-only>
+          <RadioPlayer :show-album-art="true" :now-playing-uri="streamServer" />
+        </client-only>
+      </div>
     </div>
-    <div class="flex items-center cursor-pointer" @click="$router.push('/')">
-      <img src="/img/lahmacun-logo.png" alt="" class="h-16 mr-4">
-      <h1>Lahmacun Radio</h1>
+    <HeadNavigation />
+    <div id="time">
```
Comment (pvj) [inline components/HeaderBlock.vue:14]: I like your logic of having HeaderBlock > HeaderNavigation 

Is there a reason time is not a component of its own?

File: components/ScheduleHome.vue [hunk 13]
```diff
@@ -1,15 +1,26 @@
 <template>
```
Comment (pvj) [inline components/ScheduleHome.vue:1]: This is just me thinking out loud maybe this is stupid thing to ask or assume in Vue-world. 

I know we display this schedule on the home page. But I believe the template will be the same on the schedule page daily view as well.

So do you think we could re-use this component there as well? 

Because if we can I would rename the component to something like: ScheduleDailyList

And then this component can be used on `/` and also on `/schedule/list` too without any modification? At least this is my dream-like scenario. Then even if you switch the days on `/schedule/list` page from today to tomorrow or 2028. 11. 04. we would use this component for rendering that as well just w/ different data.

And you could just add `ScheduleWeeklyList` or `ScheduleDailyGrid` or `ScheduleWeeklyGrid` etc. for the different displays and designs -- maybe there is some `ScheduleAbstract` component that is used by all these more custom components (I'm not sure I know enough about inheritance in Vue-world).

Again, sorry if this is the wrong train of thought but I wanted to mention these ideas.

File: components/ScheduleHome.vue [hunk 14]
```diff
@@ -1,15 +1,26 @@
...
       </div>
-      <div class="dayshows">
-        <div v-for="show in day" :key="show.id" :class="showAirCheck(index, show.name) ? 'dayblock onair' : 'dayblock'">
+      <div class="infos day">
+        <h3>{{ todayName }}</h3>
+      </div>
+    </div>
+    <div class="flex items-center scheduleblock">
+      <div class="w-full m-auto">
+        <div v-for="show in showsByDate[0]" :key="show.id" :class="showAirCheck(0, show.name) ? 'dayblock onair' : 'dayblock'">
```
Comment (pvj) [inline components/ScheduleHome.vue:15]: Maybe later on we should add a nice error message if there is no data?

File: components/ScheduleHome.vue [hunk 15]
```diff
@@ -21,6 +32,29 @@
...
     </div>
+    <div v-if="false" class="old-schedule">
+      <div v-for="(day, index) in showsByDate" :key="day.id" class="mb-2 scheduleblock">
+        <div class="flex flex-row justify-between dayname">
+          <h4>{{ dayNames[day[0].day-1] }}</h4>
+          <span>{{ format(new Date(index === 0 ? todayDate :
+            tommorrow.setDate(tommorrow.getDate() + 1)) , 'MMMM dd.') }}</span>
+        </div>
+        <div class="dayshows">
+          <div v-for="show in day" :key="show.id" :class="showAirCheck(index, show.name) ? 'dayblock onair' : 'dayblock'">
```
Comment (pvj) [inline components/ScheduleHome.vue:43]: This is an awesome solution! So cool that you can evaluate this `showAirCheck()` function for deciding which class to use

File: components/ScheduleHome.vue [hunk 16]
```diff
@@ -21,6 +32,29 @@
...
+            <div class="show-basic-infos">
+              {{ removeSeconds(show.start) }}-{{ removeSeconds(show.end) }} -
```
Comment (pvj) [inline components/ScheduleHome.vue:45]: Cool!

File: components/ScheduleHome.vue [hunk 17]
```diff
@@ -21,6 +32,29 @@
...
+              <nuxt-link :to="'/shows/' + show.archive_lahmastore_base_url">
```
Comment (pvj) [inline components/ScheduleHome.vue:46]: Wow!! Cool!!! This is honestly cool to see applied like this

Comment (pvj) [reply components/ScheduleHome.vue:46]: Makes me think I should've found a better name for this field

File: components/ScheduleHome.vue [hunk 18]
```diff
@@ -21,6 +32,29 @@
...
+                {{ show.name }}
+              </nuxt-link>
+              <div class="show-image" :style="{ backgroundImage: `url(${show.cover_image_url})` }" />
```
Comment (pvj) [inline components/ScheduleHome.vue:49]: Wow! Awesome! I was just thinking about adding the show cover image for each show in the schedule

File: components/ScheduleHome.vue [hunk 19]
```diff
@@ -21,6 +32,29 @@
...
+            </div>
+            <div v-if="showAirCheck(index, show.name)" class="onairtext">
```
Comment (pvj) [inline components/ScheduleHome.vue:51]: Maybe if the show is on air we could bold the text as well? Or display it w/ bigger font? Or would that be too much of an effect? Maybe it is better to be subtle. Well anyway this is definitely not for me to decide

File: components/ScheduleHome.vue [hunk 20]
```diff
@@ -64,6 +98,9 @@ export default {
       } else {
         return this.nowPlaying?.now_playing?.song.artist
```
Comment (pvj) [inline components/ScheduleHome.vue:99]: This conditional operator is actually pretty cool!  

But if it returns none then how do we handle it? Since it is already in the `else` statement part of execution

File: components/news/Home.vue [hunk 21]
```diff
@@ -0,0 +1,115 @@
...
+      <div class="news-date">
+        <div class="newsgeneral">
+          <nuxt-link to="news/">
+            <h3 class="m-0">
+              News
+            </h3>
+          </nuxt-link>
+        </div>
+        <div class="newsdate">
+          <p>{{ format(new Date(news.modified), 'MM') }}</p>
```
Comment (pvj) [inline components/news/Home.vue:18]: It seems like we are using time and date in many places. 

When you develop in js / vue what's preferred? To always just format these and create time / date objects inplace or to have generic methods somewhere and use those for all over the application? 

Sorry if this is dumb question but I don't know what the best practices are for current front-end development.

File: components/news/Home.vue [hunk 22]
```diff
@@ -0,0 +1,115 @@
...
+          <p>{{ format(new Date(news.modified), 'dd') }}</p>
+        </div>
+      </div>
+      <div class="px-4 news-info-details">
+        <h5 class="py-4">
+          <nuxt-link :to="'/news/' + news.slug">
+            {{ htmlDecoder(news.title.rendered) }}
```
Comment (pvj) [inline components/news/Home.vue:25]: I searched our repo for htmlDecoder this is 1000% amazing & smart mixin method! https://github.com/mmmnmnm/lahmacun_rework/blob/03990598f3cee4655f13c949bedfbdb55f74f6ed/plugins/mixinCommonMethods.js#L34 I think you were quite smart to handle it this way

Comment (pvj) [reply components/news/Home.vue:25]: Maybe these are not related. 

So I searched for the `htmlDecoder()` stuff and I found that you have a `news/_slug` object and `head()` method there returning the title already what is the difference between that and this, why don't we just use that?

File: components/news/Home.vue [hunk 23]
```diff
@@ -0,0 +1,115 @@
...
+          </nuxt-link>
+        </h5>
+        <div v-html="news.excerpt.rendered" />
+        <div class="flex justify-end my-4">
+          <div class="px-4 py-2 bg-gray-200 rounded-lg cursor-pointer more-button" @click="$emit('changenews')">
```
Comment (pvj) [inline components/news/Home.vue:30]: Well I was never fan and never understood the js stuff but I have to say these click events and this emit stuff is really really smooth & cool. Good job!

File: components/news/Home.vue [hunk 24]
```diff
@@ -0,0 +1,115 @@
...
+  },
+  watch: {
+    news: {
+      handler () {
+        this.loadNewsImages(this.news.featured_media)
+      }
+
+    }
+  },
+  mounted () {
```
Comment (pvj) [inline components/news/Home.vue:68]: Is it okay that `handler()` is part of the `news:` obj while `mounted()` is part of `watch:` 

Can you link me the docs or some place to understand this area of Vue (js?) better?

File: components/news/Home.vue [hunk 25]
```diff
@@ -0,0 +1,115 @@
...
+    this.loadNewsImages(this.news.featured_media)
+  },
+  methods: {
+    async loadNewsImages (newsId) {
+      const adress = `${contentApiURL}/media/${newsId}`
+      const responseNews = await this.$axios.get(adress)
+      this.newsImage = responseNews.data?.media_details?.sizes?.large?.source_url || responseNews.data?.source_url || this.newsImage
```
Comment (pvj) [inline components/news/Home.vue:75]: Cool! This means we can optimise images for example to save bandwith on mobile networks?

File: layouts/default.vue [hunk 26]
```diff
@@ -1,8 +1,7 @@
 <template>
   <div>
     <HeaderBlock />
-    <head-navigation />
-    <nuxt />
+    <nuxt class="py-8" />
```
Comment (pvj) [inline layouts/default.vue:4]: python? :D Just kidding but honestly I'm curious what is py here?

File: pages/index.vue [hunk 27]
```diff
@@ -1,23 +1,10 @@
 <template>
```
Comment (pvj) [inline pages/index.vue:1]: Can you draw a tree or explain with `>` `<` relation marks how to think about `layouts/` vs. `pages/`? 

I see we have `default` and `home` layouts and I think I understand that those are the templates and then `pages/` are the actual views where you combine templates w/ data?

File: pages/index.vue [hunk 28]
```diff
@@ -26,15 +13,13 @@
 
 <script>
 export default {
-  components: {
-  },
+  layout: 'home',
   data () {
     return {
-      newsLimit: 3
+      newsLimit: 9,
```
Comment (pvj) [inline pages/index.vue:19]: Nice!

File: pages/index.vue [hunk 29]
```diff
@@ -53,14 +38,19 @@ export default {
     }
   },
   methods: {
+    changenews () {
```
Comment (pvj) [inline pages/index.vue:41]: nice method! 

Why do you subtract 1 why don't we let `newsStart` become the same number as `newsLimit`? Is this an index starting from 0?

**Review by pvj — APPROVED** [review]
I approve based on the results this is tremendous boost to our development efforts. I don't even know how you had the time but anyway you did it. 

Great job Gas! One of thebiggest changesets I had to review in quite some time and in language / framework / discipline I don't know much about. Sorry if I made your life harder!

File: components/ScheduleHome.vue [hunk 30]
```diff
@@ -21,6 +32,29 @@
         </div>
       </div>
     </div>
+    <div v-if="false" class="old-schedule">
```
[the remark this answers is not in the document]
Comment (baxgas) [reply components/ScheduleHome.vue:35]: I just left the code for the old schedule invisible, I think we'll need some chunks from it eg. to show the other days (other than current)

File: pages/index.vue (hunk 1 above)
Comment (baxgas) [reply pages/index.vue:5]: yes, folder/names can be used also

File: assets/css/_variables.scss [hunk 31]
```diff
@@ -0,0 +1,3 @@
+$lahma-pink: #d09cf8;
+$black-color: #23282d;
+$lahma-type: 'Rubik', "Segoe UI","Helvetica Neue",Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol"
```
[the remark this answers is not in the document]
Comment (baxgas) [reply assets/css/_variables.scss:3]: She used the same font and lahma-pink, but sure, she can decide about this

File: components/FooterHome.vue [hunk 32]
```diff
@@ -0,0 +1,93 @@
+<template>
```
[the remark this answers is not in the document]
Comment (baxgas) [reply components/FooterHome.vue:1]: for now I separated the layout of the home page, if we use this footer everywhere I'll rename it definately

File: components/FooterHome.vue [hunk 33]
```diff
@@ -0,0 +1,93 @@
...
+            <img src="/img/rogue-amoeba-logo.png" alt="">
+          </a>
+        </div>
+      </div>
+    </div>
+  </footer>
+</template>
+
+<script>
+export default {
```
[the remark this answers is not in the document]
Comment (baxgas) [reply components/FooterHome.vue:46]: it's empty for now, we'll fill it up if there will be some js needed

File: components/FooterHome.vue [hunk 34]
```diff
@@ -0,0 +1,93 @@
...
+
+}
+</script>
+
+<style lang="scss" scoped>
```
[the remark this answers is not in the document]
Comment (baxgas) [reply components/FooterHome.vue:51]: yes, scoped styles apply only to the current component, not to leak out to the general styling (there is globalstyles.scss for general styling)

File: components/HeadNavigation.vue [hunk 35]
```diff
@@ -1,33 +1,34 @@
...
+          About Us
         </NuxtLink>
       </li>
-      <li>Photos</li>
-      <li>Sponsors</li>
       <li>
-        <NuxtLink to="/about/">
-          About Us
+        <NuxtLink to="/shows/">
+          Shows
```
[the remark this answers is not in the document]
Comment (baxgas) [reply components/HeadNavigation.vue:11]: Good idea. Can be set somewhere yes, but Nuxt/Vue doesn't provide admin interface in itself. Maybe in Arcsi, or wp there could be some endpoint with json objects name, URL etc. ?

File: components/HeaderBlock.vue (hunk 2 above)
Comment (baxgas) [reply components/HeaderBlock.vue:2]: it's tailwind css classes for styling, similar to bootstrap (but much nicer and modern)

File: components/HeaderBlock.vue (hunk 3 above)
Comment (baxgas) [reply components/HeaderBlock.vue:9]: this is the way the current player works also. but any mp3 data stream should work with it yes. we could have some prop also eg. :arcsi-play="true" and handle all the differences between stream/arcsi with it
I was also thinking to have a unified player, we can discuss it on Saturday.

File: components/HeaderBlock.vue (hunk 4 above)
Comment (baxgas) [reply components/HeaderBlock.vue:16]: yes, lahmacun time is important because of the schedule (eg. the Gestalt dude has different time-zone)
a question for Saturday?

File: components/HeaderBlock.vue (hunk 6 above)
Comment (baxgas) [reply components/HeaderBlock.vue:46]: hm, good idea for the crosshair. also topic for Saturday?
`getNow` can be a mixin function, whynot?

File: components/RadioPlayer.vue (hunk 7 above)
Comment (baxgas) [reply components/RadioPlayer.vue:10]: showAlbumArt is the prop we send in (above you had a comment on that line also) – you are right, it is obsolete if we will always use image.
np.now_playing.song.art was done by gammaw with all the if/else cases of availability of image, default azuracast image etc. with fallbacks to show default
nice ideas with other displays, it is also dependable if we have one player for stream/arcsi if we show the chats there... topic for Saturday

File: components/RadioPlayer.vue (hunk 8 above)
Comment (baxgas) [reply components/RadioPlayer.vue:15]: @close is the built in method for modals (very clever), it triggers a function in the Modal component

File: components/RadioPlayer.vue (hunk 9 above)
Comment (baxgas) [reply components/RadioPlayer.vue:18]: good idea, but first let's decide if arcsi player is also here

File: components/RadioPlayer.vue (hunk 10 above)
Comment (baxgas) [reply components/RadioPlayer.vue:30]: this is not modal, but the volume slider, same as the current player on prod lahma

File: components/RadioPlayer.vue (hunk 11 above)
Comment (baxgas) [reply components/RadioPlayer.vue:36]: .prevent is preventing the browser triggering the default click event, same as event.preventDefault
mute is also possible now, it was implemented from the beginning :-)

File: components/HeaderBlock.vue (hunk 12 above)
Comment (baxgas) [reply components/HeaderBlock.vue:14]: could be a component on it's own for sure

File: components/ScheduleHome.vue (hunk 13 above)
Comment (baxgas) [reply components/ScheduleHome.vue:1]: yes, good idea for further developing the structure and UI. for now it is also a good question how and where to show the other days on home page, I would think some dropdown for selecting other days, or scrolling left-right
the other Schedule view is a nice addition to the rework, definitely worth the effort to develop it, now that technically everything is under our fingertips. topic for Saturday!

File: components/ScheduleHome.vue (hunk 14 above)
Comment (baxgas) [reply components/ScheduleHome.vue:15]: no data???

File: components/ScheduleHome.vue (hunk 18 above)
Comment (baxgas) [reply components/ScheduleHome.vue:49]: not the nicest solution, but it works, and I was really amazed to see that vue is loading these images on-demand and therefore reducing the bandwidth strain

Comment (baxgas) [reply components/ScheduleHome.vue:49]: we can discuss this further on Saturday how this hover image show is the nicest, and maybe also to add some other meta info to this popup

File: components/ScheduleHome.vue (hunk 19 above)
Comment (baxgas) [reply components/ScheduleHome.vue:51]: question for Csenge, let her work also :-)

File: components/ScheduleHome.vue (hunk 20 above)
Comment (baxgas) [reply components/ScheduleHome.vue:99]: it is the new Nullish coalescing operator, really makes things easier for handling unavailable data.
if none is returning it just gives empty string I think, but if you can think some edge cases feel free to tell

File: components/news/Home.vue (hunk 21 above)
Comment (baxgas) [reply components/news/Home.vue:18]: format is a part of date-fns library for showing the time in different format https://date-fns.org/v2.22.1/docs/format
new Date is a js Date object

File: components/news/Home.vue (hunk 22 above)
Comment (baxgas) [reply components/news/Home.vue:25]: htmlDecoder fixes the errors from wordpress json with some extra characters sent as HTML entity codes. you can check what it does if you turn it off in news titles

File: components/news/Home.vue (hunk 23 above)
Comment (baxgas) [reply components/news/Home.vue:30]: $emit emits from children to parent component

File: components/news/Home.vue (hunk 24 above)
Comment (baxgas) [reply components/news/Home.vue:68]: watcher is a vue way of checking if something is https://vuejs.org/v2/guide/computed.html#Watchers

File: components/news/Home.vue (hunk 25 above)
Comment (baxgas) [reply components/news/Home.vue:75]: wp makes different image sizes on upload, thumbnail, medium and large

File: layouts/default.vue (hunk 26 above)
Comment (baxgas) [reply layouts/default.vue:4]: py-8 = padding on y-axis with 8 units (8*0.25rem)

File: pages/index.vue (hunk 27 above)
Comment (baxgas) [reply pages/index.vue:1]: layouts are different layouts of components, that can be set in pages -> pages use the default layout by default, but you can change the layout to some custom via this property

File: pages/index.vue (hunk 29 above)
Comment (baxgas) [reply pages/index.vue:41]: yes, the index starts from zero, that' why we need -1 for the total number (newsLimit)

File: components/ScheduleHome.vue (hunk 30 above)
[the remark this answers is not in the document]
Comment (gammaw) [reply components/ScheduleHome.vue:35]: I see. How about commenting it out then?

**Comment by baxgas** [comment]
V-if=false is a commented out block. But true, can be removed all
together,since we can get it back via github history

File: pages/index.vue (hunk 27 above)
Comment (pvj) [reply pages/index.vue:1]: That's good explanation in layman's terms thanks bax!

File: layouts/default.vue (hunk 26 above)
Comment (pvj) [reply layouts/default.vue:4]: So there's also my and px and mx? Thanks!

File: components/ScheduleHome.vue (hunk 20 above)
Comment (pvj) [reply components/ScheduleHome.vue:101]: You use it in `showAirCheck` for logic to verify there is some show on air. That is why I asked how we handle it. Is an empty string ~trueish?

File: components/ScheduleHome.vue (hunk 19 above)
Comment (pvj) [reply components/ScheduleHome.vue:52]: Agreed and discussed in person!

File: components/ScheduleHome.vue (hunk 18 above)
Comment (pvj) [reply components/ScheduleHome.vue:50]: This was one of my favourites during the demo. Great idea & implementation! 

Vue is cool too they really played their part in this based on what you're writing! 

I would like to have more meta info in there for sure!

File: components/ScheduleHome.vue (hunk 14 above)
Comment (pvj) [reply components/ScheduleHome.vue]: Yeah maybe there's no feed coming from back-end or something. But it can be revisited later in focus

File: components/RadioPlayer.vue (hunk 11 above)
Comment (pvj) [reply components/RadioPlayer.vue:36]: Yeah but not w/ click like that at least I don't remember it. Sorry if it has been there all along!

File: components/RadioPlayer.vue (hunk 10 above)
Comment (pvj) [reply components/RadioPlayer.vue:30]: Do you think the volume slider could be part of the player modal?

File: components/RadioPlayer.vue (hunk 8 above)
Comment (pvj) [reply components/RadioPlayer.vue:15]: Thanks for the teachment

File: components/RadioPlayer.vue (hunk 7 above)
Comment (pvj) [reply components/RadioPlayer.vue:10]: Good studiment, thank you! Yeah I vaguely remember the now playing logic this will be lot cleaner.
